lusidtools/cocoon/instruments.py

```diff
@@ -292,17 +292,28 @@ def get_unique_identifiers(api_factory: lusid.utilities.ApiClientFactory):
         if identifier.is_unique_identifier_type
     ]
 
+def get_leaf(d, IDs):
+    for lusidID, dataID in d.items():
+        if isinstance(dataID, dict) and "column" not in dataID:
+            get_leaf(dataID, IDs)
+        elif isinstance(dataID, dict) and "column" in dataID:
+            IDs.append((lusidID, dataID["column"]))
+        elif dataID[0] != "$":
+            IDs.append((lusidID, dataID))
```

Comment:
If this is looking for a constant prefix, that prefix should be a function argument not hardcoded to be "$".  You can see that it is already passed into enrich_instruments as the argument constant_prefix.

Reply:
added constant_prefix as arg to get_leaf()